src/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -84,7 +92,43 @@ public RestoreCommand()
             }
         }
 
-        private async Task PerformNuGetV3RestoreAsync(string projectPath)
+        private static string GetEffectiveGlobalPackagesFolder(
+            string packagesDirectoryParameter,
+            string solutionDirectoryParameter,
+            PackageRestoreInputs packageRestoreInputs,
+            string globalPackagesFolder)
+        {
+            if (!string.IsNullOrEmpty(packagesDirectoryParameter))
+            {
+                return packagesDirectoryParameter;
+            }
+
+            if (!string.IsNullOrEmpty(solutionDirectoryParameter) || packageRestoreInputs.RestoringWithSolutionFile)
```

Comment:
do the ifs that setup the solutionDirectory (instead of if and a nested trinary) and then do an if checking the solution directory.